File: csharp/sones-GraphMappingFramework/sones-GraphMappingFramework/DBObjects/EdgeType/IEdgeType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using sones.GraphMappingFramework.DBObjects.Definitions;

namespace sones.GraphMappingFramework.DBObjects.EdgeType
{
    public interface IEdgeType
    {
         Boolean IsSealed();
         Boolean HasParentType();
         Boolean HasChildTypes();
         Boolean IsAncestor(IEdgeType myotherType);
         Boolean IsAncestor(IEdgeType myotherType, Boolean myIncludeSelf);
         Boolean IsDescendant(IEdgeType myotherType);
         Boolean IsDescendant(IEdgeType myotherType, Boolean myIncludeSelf);
         List<IEdgeType> GetDescendantTypes();
         List<IEdgeType> GetDescendantTypes(Boolean myIncludeSelf);
         List<IEdgeType> GetAncestorTypes();
         List<IEdgeType> GetAncestorTypes(Boolean myIncludeSelf);
         List<IEdgeType> GetKinsmenTypes();
         List<IEdgeType> GetKinsmenTypes(Boolean myIncludeSelf);
         List<IEdgeType> GetChildrenTypes();
         List<IEdgeType> GetParentType();

        #region Attributes

         Boolean HasAttribute(String myAttributeName);
         Boolean HasAttributes(Boolean myIncludeAncestors);
         AttributeDefinition GetAttributeDefinition(String myAttributeName);
         List<AttributeDefinition> GetAttributeDefinitions(Boolean myIncludeAncestors);

        #endregion
        
        #region Properties

         Boolean HasProperty(String myPropertyName);
         Boolean HasProperties(Boolean myIncludeAncestors);
         PropertyDefinition GetPropertyDefinition(String myPropertyName);
         List<PropertyDefinition> GetPropertyDefinitions(Boolean myIncludeAncestors);
         List<PropertyDefinition> GetPropertyDefinitions(List<String> myPropertyNames);

        #endregion
    }
}

File: java/sones-GraphMappingFramework/src/sones/GraphMapping/DBObjects/IEdgeType.java
package sones.GraphMapping.DBObjects;

import java.util.List;

import sones.GraphMapping.Definitions.AttributeDefinition;
import sones.GraphMapping.Definitions.PropertyDefinition;

public interface IEdgeType {
	public boolean IsSealed();
	public boolean HasParentType();
	public boolean HasChildTypes();
	public boolean IsAncestor(IEdgeType _otherType);
	public boolean IsAncestor(IEdgeType _otherType, boolean _IncludeSelf);
	public boolean IsDescendant(IEdgeType _otherType);
	public boolean IsDescendant(IEdgeType _otherType, boolean _IncludeSelf);
	public List<IEdgeType> GetDescendantTypes();
	public List<IEdgeType> GetDescendantTypes(boolean _IncludeSelf);
	public List<IEdgeType> GetAncestorTypes();
	public List<IEdgeType> GetAncestorTypes(boolean _IncludeSelf);
	public List<IEdgeType> GetKinsmenTypes();
	public List<IEdgeType> GetKinsmenTypes(boolean _IncludeSelf);
	public List<IEdgeType> GetChildrenTypes();
	public List<IEdgeType> GetParentType();
	// Attributes
	public boolean HasAttribute(String _AttributeName);
	public boolean HasAttributes(boolean _IncludeAncestors);
	public AttributeDefinition GetAttributeDefinition(String _AttributeName);
	public List<AttributeDefinition> GetAttributeDefinitions(boolean _IncludeAncestors);
	// Properties
	public boolean HasProperty(String _PropertyName);
	public boolean HasProperties(boolean _IncludeAncestors);
	public PropertyDefinition GetPropertyDefinition(String _PropertyName);
	public List<PropertyDefinition> GetPropertyDefinitions(boolean _IncludeAncestors);
	public List<PropertyDefinition> GetPropertyDefinitions(List<String> _PropertyNames);
}
